### PR TITLE
Min max to dB and simplified dualslider

### PR DESCRIPTION
**Pull request by 777arc**

### CODE REVIEW

**Review by Nepomuceno — APPROVED** [review]
WOW just love this ❤️ reducingg a lot of the most complicated part of the code for me :)

**Comment by 777arc** [comment]
@Jaydee123 what do you think about those changes to your dualslider, having it no longer have its own state?  Seems to work fine, at least with this use-case.

**Comment by Jaydee123** [comment]
> @Jaydee123 what do you think about those changes to your dualslider, having it no longer have its own state? Seems to work fine, at least with this use-case.

My only reservations is that a 'Dual Slider' component is meant to be generic, by referring to the 'Magnitude' we're tying it to that particular scenario, can we name it to 'updateMin' inside the class so it has more of a generic name?

File: client/src/Components/DualRangeSlider/DualRangeSlider.tsx [hunk 1]
```diff
@@ -1,65 +1,64 @@
-import React, { useCallback, useEffect, useState, useRef } from 'react';
+import React, { useCallback, useEffect, useRef } from 'react';
 
-const DualRangeSlider = ({ min, minValue, max, maxValue, onChangeMin, onChangeMax }) => {
-  const [minVal, setMinVal] = useState(minValue);
-  const [maxVal, setMaxVal] = useState(maxValue);
+const DualRangeSlider = ({ min, minValue, max, maxValue, updateMagnitudeMin, updateMagnitudeMax }) => {
```
Comment (Nepomuceno) [inline client/src/Components/DualRangeSlider/DualRangeSlider.tsx:3]:
```suggestion
const DualRangeSlider = ({ min, minValue, max, setValue, setMin, setMax }) => {
```
change this name would make the componetn a bit more generic.

File: client/src/Components/DualRangeSlider/DualRangeSlider.tsx [hunk 2]
```diff
@@ -1,65 +1,64 @@
...
-  const [maxVal, setMaxVal] = useState(maxValue);
+const DualRangeSlider = ({ min, minValue, max, setValue, setMin, setMax }) => {
   const range = useRef(null);
-
   const getPercent = useCallback((value) => Math.round(((value - min) / (max - min)) * 100), [min, max]);
 
-  useEffect(() => {
-    const minPercent = getPercent(minVal);
-    const maxPercent = getPercent(maxVal);
+  // if some other component changes the max/min, update the slider visually
```
Comment (Jaydee123) [inline client/src/Components/DualRangeSlider/DualRangeSlider.tsx:7]: can we remove

File: client/src/Components/DualRangeSlider/DualRangeSlider.tsx [hunk 3]
```diff
@@ -1,65 +1,64 @@
...
+    const minPercent = getPercent(minValue);
+    const maxPercent = getPercent(maxValue);
     if (range.current) {
       range.current.style.left = `${minPercent}%`;
       range.current.style.width = `${maxPercent - minPercent}%`;
     }
-
-    onChangeMin(minVal);
-  }, [minVal, getPercent]);
+    updateMagnitudeMin(minValue);
```
Comment (Jaydee123) [inline client/src/Components/DualRangeSlider/DualRangeSlider.tsx:20]: needs changing to setMin

File: client/src/Components/DualRangeSlider/DualRangeSlider.tsx [hunk 4]
```diff
@@ -1,65 +1,64 @@
...
-
+    const minPercent = getPercent(minValue);
+    const maxPercent = getPercent(maxValue);
     if (range.current) {
       range.current.style.width = `${maxPercent - minPercent}%`;
     }
-
-    onChangeMax(maxVal);
-  }, [maxVal, getPercent]);
+    updateMagnitudeMax(maxValue);
```
Comment (Jaydee123) [inline client/src/Components/DualRangeSlider/DualRangeSlider.tsx:29]: needs chaning to setMax

File: client/src/Components/DualRangeSlider/DualRangeSlider.tsx [hunk 5]
```diff
@@ -1,65 +1,64 @@
...
         type="range"
         min={min}
         max={max}
-        value={minVal}
+        value={minValue}
         onChange={(event) => {
-          const value = Math.min(Number(event.target.value), maxVal - 1);
-          setMinVal(value);
+          const value = Math.min(Number(event.target.value), maxValue - 1);
+          updateMagnitudeMin(value);
```
Comment (Jaydee123) [inline client/src/Components/DualRangeSlider/DualRangeSlider.tsx:41]: same here

File: client/src/Components/DualRangeSlider/DualRangeSlider.tsx [hunk 6]
```diff
@@ -1,65 +1,64 @@
...
         type="range"
         min={min}
         max={max}
-        value={maxVal}
+        value={maxValue}
         onChange={(event) => {
-          const value = Math.max(Number(event.target.value), minVal + 1);
-          setMaxVal(value);
+          const value = Math.max(Number(event.target.value), minValue + 1);
+          updateMagnitudeMax(value);
```
Comment (Jaydee123) [inline client/src/Components/DualRangeSlider/DualRangeSlider.tsx:52]: same here

**Pull request merged**